### PR TITLE
feat(v05): T-V05-013 strip-and-stubify automation

### PR DESCRIPTION
**Pull request by Luis85**
## Summary

- **Resolves DEFECT-V05-001** from issue #90 §1 — the v0.5 release-blocker raised by PR #162's Stage 8 testing pass (22 numbered ADRs + multiple built-up docs shipping in the candidate archive). After this PR the candidate archive passes Layer 2 fresh-surface readiness with zero diagnostics.
- **Closes OQ-V05-003 fully.** The strip-and-stubify packaging step is now automated; maintainers no longer strip manually using `docs/release-package-contents.md` as a checklist.
- **Adds T-V05-013** as the new dev-owned task to `specs/version-0-5-plan/tasks.md`.

## Why the build-time-transform path

Issue #90 §1 named the resolution as "Convert all shipping `docs/` pages to stub form per `templates/release-package-stub.md`". Reading that as a direct rewrite of codebase docs is internally inconsistent: `docs/release-package-contents.md` line 40 says "the codebase form remains as authored — maintainers reading the codebase see one shape, consumers receive another." The build-time transform is the only path that preserves both surfaces.

`design.md` §"Build-time transform pattern (T-V05-013)" records the architecture and the rationale for the staged-tree shape over a `prepack` lifecycle hook (which would either need a paired `postpack` revert or commit stub form into the codebase).

## What changed

**New scripts**
- `scripts/lib/release-stubify.ts` — pure single-doc transform. Synthesises frontmatter from path, preserves `# ` and `## ` headings, replaces bodies with `<!-- TODO:` markers, drops `### ` and deeper, ignores fenced code blocks for heading detection, normalises CRLF. Stable exports `STUB_FRONTMATTER_KEYS` and `STUB_TODO_MARKER` mirror the constants the Layer 2 check enforces, so any drift between transform output and assertion 3 fails tests.
- `scripts/lib/release-archive-builder.ts` — orchestrator over the canonical `package.json#files` allowlist. Classifier returns `skip` for numbered ADRs, `copy` for the stub template itself (it _is_ the reference shape), `stubify` for every other `docs/**/*.md`, and `copy` for everything else. Idempotent across re-runs; rejects paths that escape the repo root.
- `scripts/build-release-archive.ts` — CLI. Enumerates the file allowlist via `npm pack --dry-run --json` from the repo root (single source of truth: `package.json#files`), then dispatches each path to `buildReleaseArchive`. `--out <dir>`, `--no-clean`, `--json` flags.

**Workflow wiring**
- `.github/workflows/release.yml` step 5 now runs `cp package-lock.json npm-shrinkwrap.json` → `npm run build:release-archive -- --out .release-staging` → `npm pack ./.release-staging`. The tarball reflects the released form; Layer 2 readiness in step 6 asserts the contract against the extracted candidate.
- Step 10 publishes the byte-identical staged tarball via `npm publish "${TARBALL}"` so the GitHub Packages version equals the GitHub Release asset attached in step 11. Publishing from the codebase form via bare `npm publish` would have bypassed the transform entirely.

**Defence-in-depth**
- `.npmignore` excludes `docs/adr/[0-9][0-9][0-9][0-9]-*.md`, `.worktrees/`, and `.release-staging/` so a maintainer running `npm pack` directly from the repo root (bypassing the build script) does not accidentally publish a built-up archive.
- `.gitignore` excludes `.release-staging/` and the runner-only `npm-shrinkwrap.json`.

**Spec artifacts**
- `specs/version-0-5-plan/tasks.md` — T-V05-013 added.
- `specs/version-0-5-plan/design.md` — new "Build-time transform pattern" subsection + two affected-surface rows.
- `specs/version-0-5-plan/implementation-log.md` — T-V05-013 entry.
- `specs/version-0-5-plan/test-report.md` §5 — DEFECT-V05-001 marked resolved on the technical surface.
- `specs/version-0-5-plan/package-contract.md` §9 — OQ-V05-003 closed; change-log entry.
- `specs/version-0-5-plan/workflow-state.md` — hand-off note.
- `tools/automation-registry.yml` — `build:release-archive` entry added.
- `docs/scripts/` — typedoc shells regenerated.

## Test plan

- [x] 19 unit tests on `scripts/lib/release-stubify.ts` (frontmatter shape, title derivation paths, folder derivation, description fallback, `entry_point` preservation, section heading preservation, code-block heading guard, CRLF, missing-section edge case, etc.).
- [x] 13 unit tests on `scripts/lib/release-archive-builder.ts` (classifier per shape; mirror non-docs as-is; stubify docs; skip numbered ADRs; end-to-end build → Layer 2 check passes; intermediate dir creation; idempotency; written/skipped report; escape-path rejection).
- [x] `npm run verify` green across all 18 gates (241 tests pass = 209 baseline + 32 new).
- [x] End-to-end: `npm run build:release-archive` produces a 826-file staging dir with 22 numbered ADRs filtered. `npm pack ./.release-staging` produces a tarball whose extracted contents pass Layer 2 with zero diagnostics.
- [ ] CI: actionlint + zizmor on the workflow change.
- [ ] CI: full verify + link checks.

## Out of scope (still needed before v0.5.0 ships)

- **CLAR-V05-003** — whether the first publish should be draft / pre-release only. Owner: human decision.
- **Operator-authorised remote `gh workflow run release.yml` dispatch.** Article IX requires explicit human authorisation.
- **Stage 9 `/spec:review`** and **Stage 11 `/spec:retro`** for `specs/version-0-5-plan/`.

This PR clears the technical blocker. The remaining pickup items in issue #90 are decisions or operator-authorised actions, not dev work.

Refs: #90, T-V05-013, DEFECT-V05-001, OQ-V05-003, SPEC-V05-010, ADR-0021, REQ-V05-005, REQ-V05-012

### CODE REVIEW

File: scripts/build-release-archive.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+let parsed: ParsedArgs;
+try {
+  parsed = parseArgs(process.argv.slice(2));
+} catch (err) {
+  console.error(`${heading}: ${(err as Error).message}`);
+  process.exit(1);
+}
+
+if (parsed.cleanFirst && fs.existsSync(parsed.outDir)) {
+  fs.rmSync(parsed.outDir, { recursive: true, force: true });
```
[the remark this answers is not in the document]
Comment (Luis85) [reply scripts/build-release-archive.ts]: Addressed in commit `b13a57e`. New `scripts/lib/release-staging-safety.ts` exports `assertSafeOutDir(outDir, repoRoot)` called before `fs.rmSync` at `scripts/build-release-archive.ts:113`. Rejects (in order): filesystem root, user home, repo root, repo parent, any ancestor of the repo root, an existing path that is not a directory, an existing directory containing a `.git` entry, and any non-empty existing directory that lacks the `.release-staging-marker` file. Empty directories, non-existent paths, and previously-staged dirs (marker present) all pass. 13 new unit tests cover every branch (`tests/scripts/release-staging-safety.test.ts`).

File: .npmignore [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# - `package.json#files` (the canonical include list).
+# - `scripts/lib/release-archive-builder.ts` (build-time filter that produces
+#   the staged tree).
+# - `docs/release-package-contents.md` (the contract documentation).
+# - `tools/automation-registry.yml` (the automation inventory).
+
+# Numbered ADRs ship in the codebase only — the consumer's first ADR is
+# their own ADR-0001, not a continuation of ours (ADR-0021 §Decision.2,
+# SPEC-V05-010 assertion 1).
+docs/adr/[0-9][0-9][0-9][0-9]-*.md
```
[the remark this answers is not in the document]
Comment (Luis85) [reply .npmignore]: Right — `package.json#files` precedence over top-level `.npmignore` makes the round-1 ADR exclusion a false defence. Addressed in commit `b13a57e`: dropped the `docs/adr/[0-9][0-9][0-9][0-9]-*.md` line and rewrote the header to call out the precedence rule. The actual defence is now `scripts/release-prepack-guard.mjs` (plain ESM so `npm pack ./.release-staging` runs it under bare `node` — the staged tree has no `node_modules` so `tsx` is unavailable), wired as `package.json#scripts.prepack`. The guard reads `package.json` at cwd; if `name !== "@luis85/agentic-workflow"` it exits 0 (no-op for downstream forks), else it requires the `.release-staging-marker` file at cwd. The marker is written by `build:release-archive` at the staged root, so `npm pack ./.release-staging` (which runs prepack from the staged dir's cwd) passes, but bare `npm pack` from the repo root fails closed before any tarball is produced. Marker is not in `package.json#files` so it does not ship in the tarball. 5 new spawn-based unit tests cover the guard logic (`tests/scripts/release-prepack-guard.test.ts`).

File: scripts/build-release-archive.ts [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+}
+
+type NpmPackEntry = { path?: string };
+type NpmPackOutput = Array<{ files?: NpmPackEntry[] }>;
+
+function enumerateFilesViaNpmPack(): string[] {
+  // Static literal command, no user input — safe to run via the platform shell
+  // so the npm `.cmd` shim resolves on Windows without an EINVAL spawn from
+  // execFileSync.
+  const stdout = execSync("npm pack --dry-run --json", {
```
[the remark this answers is not in the document]
Comment (Luis85) [reply scripts/build-release-archive.ts]: Real bug — round-2 introduced a chicken-and-egg dependency. Fixed in commit `208a200` by passing `--ignore-scripts` to the inner `npm pack --dry-run --json` enumeration call only. Bare `npm pack` from the repo root (no flag) still fails closed via the guard, so the round-2 defense is preserved. Verified locally end-to-end: (1) `npm run build:release-archive` produces an 826-file staged tree, (2) `npm pack ./.release-staging` (marker present in staged dir's cwd) succeeds, (3) Layer 2 `check:release-package-contents` against the extracted candidate exits 0 with zero diagnostics, (4) bare `npm pack` from repo root errors with the guard's helpful message pointing at `npm run build:release-archive && npm pack ./.release-staging`. Inline comment in `enumerateFilesViaNpmPack` records WHY `--ignore-scripts` is required so a future refactor cannot silently drop it.

File: package.json [hunk 4]
```diff
@@ -75,6 +75,8 @@
     "check:token-budget": "tsx scripts/check-token-budget.ts",
     "check:release-package-contents": "tsx scripts/check-release-package-contents.ts",
     "check:release-readiness": "tsx scripts/check-release-readiness.ts",
+    "build:release-archive": "tsx scripts/build-release-archive.ts",
+    "prepack": "node scripts/release-prepack-guard.mjs",
```
[the remark this answers is not in the document]
Comment (Luis85) [reply package.json:79]: Right — round-2 prepack guard would have blocked the §7.1 recovery runbook. Fixed in commit `4b4d009`. Rewrote §7.1 + §9 quick-reference: `npm ci` → `cp package-lock.json npm-shrinkwrap.json` → `npm run build:release-archive -- --out .release-staging` → `TARBALL=$(npm pack --silent ./.release-staging)` → `npm publish "${TARBALL}"`. Byte-equivalent to workflow step 5 + step 10. Inline comments in both blocks call out *why* bare `npm pack` from the repo root is blocked. Note: the runbook was already misleading even before the round-2 guard — bare `npm pack` would have shipped the 22 numbered ADRs in violation of ADR-0021 / SPEC-V05-010. The guard exposes that pre-existing inconsistency rather than introducing one.

**Comment by Luis85** [comment]
**Loop cap reached** (`feedback_pr_review_loop.md` soft cap = 3 rounds; this is round 4).

Summary of all four rounds:

| Round | Commit | Codex finding | Fix |
|---|---|---|---|
| R1 | `e336bd5` | P1 — destructive `--out` clean | `assertSafeOutDir` guard in new `scripts/lib/release-staging-safety.ts` |
| R1 | `e336bd5` | P2 — `.npmignore` precedence false defense | New `scripts/release-prepack-guard.mjs` wired as `prepack`; `.npmignore` rewrite |
| R2 | `b13a57e` | P1 — chicken-and-egg: prepack blocks build CLI | `--ignore-scripts` on inner `npm pack --dry-run --json` |
| R3 | `208a200` | P1 — operator runbook §7.1 broken by guard | Rewrote §7.1 + §9 quick-reference to use `build:release-archive` + staged-tree pack |

Each finding was a real bug, addressed end-to-end (code + tests + docs + spec artifacts). Test count: 209 baseline → 259 (50 new). Verify gate green at every push. End-to-end manually validated: build CLI → `npm pack ./.release-staging` → Layer 2 check exits 0 with zero diagnostics; bare `npm pack` from repo root fails closed via guard.

Continuing past the soft cap because all remaining loop iterations would be incremental tightening of the same surface; the underlying problem (DEFECT-V05-001 / OQ-V05-003) is resolved on the technical surface. Re-requesting Codex review on `4b4d009`. If round-4 surfaces another P1, escalating to human reviewer per the loop discipline.

File: scripts/build-release-archive.ts [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+let parsed: ParsedArgs;
+try {
+  parsed = parseArgs(process.argv.slice(2));
+} catch (err) {
+  console.error(`${heading}: ${(err as Error).message}`);
+  process.exit(1);
+}
+
+try {
+  assertSafeOutDir(parsed.outDir, repoRoot);
```
[the remark this answers is not in the document]
Comment (Luis85) [reply scripts/build-release-archive.ts]: Real semantic bug introduced by round-2 + round-4 — addressed in commit `a9adc3a`. `assertSafeOutDir` now takes `AssertSafeOutDirOptions.destructive` (default `true`). When `destructive: false` (the `--no-clean` path), only the absolute-path guards run (filesystem root, repo root, repo parent, user home, ancestors of repo root, non-directory paths). The `.git`-entry and non-empty-without-marker rejections skip because `--no-clean` is explicitly preserving the existing tree — exactly what the flag advertises. CLI passes `destructive: parsed.cleanFirst`. 4 new tests cover the non-destructive branch: passes a non-empty unmarked dir, passes a `.git`-containing dir, still rejects protected paths, still rejects a non-directory file. Test count: 263. Verify green. Also merged `origin/main` in this push (10 PRs landed since round-1) — `workflow-state.md` + `package-contract.md` conflicts resolved per `feedback_parallel_pr_conflicts.md` (merge not rebase). Round-5 is past the soft cap, but the round-4 P2 was a real semantic bug introduced by my own earlier rounds — necessary, not ping-pong.

File: scripts/build-release-archive.ts [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+  // paths. `--no-clean` then preserves whatever already lives in the dir
+  // (Codex P2 round-4 on PR #202).
+  assertSafeOutDir(parsed.outDir, repoRoot, { destructive: parsed.cleanFirst });
+} catch (err) {
+  console.error(`${heading}: ${(err as Error).message}`);
+  process.exit(1);
+}
+
+if (parsed.cleanFirst && fs.existsSync(parsed.outDir)) {
+  fs.rmSync(parsed.outDir, { recursive: true, force: true });
```
[the remark this answers is not in the document]
Comment (Luis85) [reply scripts/build-release-archive.ts]: Right — `--no-clean` is fundamentally unsafe for this build because `package.json#files` whitelists whole directories. A deleted file left in the staging tree from a prior run silently ships in the next pack. There is no safe non-destructive semantic for a deterministic mirror of the file allowlist. **Fixed in commit `4cd0df3` by dropping `--no-clean` entirely.** CLI always cleans `outDir` before staging. `cleanFirst` field removed from `ParsedArgs`. `AssertSafeOutDirOptions.destructive` reverted from the lib (round-5 added it; the only caller no longer exists). The 4 round-5 tests for the non-destructive branch are removed. The round-4 P2 fix (round-5 commit) was the wrong layer — should have been "remove the flag" not "fix the marker check". Round-6 lands the right fix; the round-5 churn is documented in the implementation log for future reference. Test count 261 (4 removed, 2 inherited from `origin/main` merge); verify green.
